dlp/src/main/java/dlp/snippets/InspectWithCustomRegex.java

```diff
@@ -0,0 +1,103 @@
+
+  public static void inspectWithCustomRegex() {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-project-id";
+    String textToInspect = "Patients MRN 444-5-22222";
+    String customRegexPattern = "[1-9]{3}-[1-9]{1}-[1-9]{5}";
+    inspectWithCustomRegex(projectId, textToInspect, customRegexPattern);
+  }
+
+  // Inspects a BigQuery Table
+  public static void inspectWithCustomRegex(
+      String projectId, String textToInspect, String customRegexPattern) {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the "close" method on the client to safely clean up any remaining background resources.
+    try (DlpServiceClient dlp = DlpServiceClient.create()) {
+      // Specify the project used for request.
+      ProjectName project = ProjectName.of(projectId);
+
+      // Specify the type and content to be inspected.
+      ByteContentItem byteItem =
+          ByteContentItem.newBuilder()
+              .setType(BytesType.TEXT_UTF8)
+              .setData(ByteString.copyFromUtf8(textToInspect))
+              .build();
+      ContentItem item = ContentItem.newBuilder().setByteItem(byteItem).build();
+
+      // Specify the regex pattern the inspection will look for.
+      Regex regex = Regex.newBuilder().setPattern(customRegexPattern).build();
+
+      // Construct the custom regex detector.
+      InfoType infoType = InfoType.newBuilder().setName("C_MRN").build();
+      CustomInfoType customInfoType =
+          CustomInfoType.newBuilder().setInfoType(infoType).setRegex(regex).build();
+
+      // Construct the configuration for the Inspect request.
+      InspectConfig config =
+          InspectConfig.newBuilder()
+              .addCustomInfoTypes(customInfoType)
+              .setIncludeQuote(true)
+              .setMinLikelihood(Likelihood.POSSIBLE)
+              .build();
+
+      // Construct the Inspect request to be sent by the client.
+      InspectContentRequest request =
+          InspectContentRequest.newBuilder()
+              .setParent(project.toString())
+              .setItem(item)
+              .setInspectConfig(config)
+              .build();
+
+      // Use the client to send the API request.
+      InspectContentResponse response = dlp.inspectContent(request);
+
+      // Parse the response and process results
+      System.out.println("Findings: " + response.getResult().getFindingsCount());
+      for (Finding f : response.getResult().getFindingsList()) {
+        System.out.println("\tQuote: " + f.getQuote());
+        System.out.println("\tInfo type: " + f.getInfoType().getName());
+        System.out.println("\tLikelihood: " + f.getLikelihood());
+      }
+    } catch (Exception e) {
+      System.out.println("Error during inspectString: \n" + e.toString());
+    }
```

Comment:
https://google.github.io/styleguide/javaguide.html#s6.2-caught-exceptions

It's better to explain what might happen, but not to catch it.